thunderbird/processes/wps_split_merged_climos.py

```diff
@@ -0,0 +1,118 @@
+# Processor imports
+from pywps import (
+    Process,
+    LiteralInput,
+    ComplexInput,
+    ComplexOutput,
+    FORMATS,
+)
+from pywps.app.Common import Metadata
+
+# Tool imports
+from nchelpers import CFDataset
+from dp.split_merged_climos import split_merged_climos
+from thunderbird.utils import (
+    MAX_OCCURS,
+    is_opendap_url,
+    collect_output_files,
+    build_meta_link,
+)
+
+# Library imports
+import logging
+import os
+import math
+
+logger = logging.getLogger("PYWPS")
+
+
+class SplitMergedClimos(Process):
+    def __init__(self):
+        self.log_level_choices = [
+            "NOTSET",
+            "DEBUG",
+            "INFO",
+            "WARNING",
+            "ERROR",
+            "CRITICAL",
+        ]
+        inputs = [
+            ComplexInput(
+                "netcdf",
+                "NetCDF Datasets",
+                abstract="NetCDF files to process",
+                min_occurs=1,
+                max_occurs=MAX_OCCURS,
+                supported_formats=[FORMATS.NETCDF, FORMATS.DODS],
+            ),
+            LiteralInput(
+                "loglevel",
+                "Log Level",
+                default="INFO",
+                abstract="Logging level",
+                allowed_values=self.log_level_choices,
+            ),
+        ]
+        outputs = [
+            ComplexOutput(
+                "output",
+                "Output",
+                abstract="Collection of split climatological files",
+                as_reference=True,
+                supported_formats=[FORMATS.META4],
+            ),
+        ]
+
+        super(SplitMergedClimos, self).__init__(
+            self._handler,
+            identifier="split_merged_climos",
+            title="Split Merged Climos",
+            abstract="Split climo means files into one file per time interval",
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+            ],
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def get_filepaths(self, request):
+        filepaths = []
+        for path in request.inputs["netcdf"]:
+            if is_opendap_url(path.url):
+                filepaths.append(path.url)
+            elif path.file.endswith(".nc"):
+                filepaths.append(path.file)
+            else:
+                raise ProcessError(
+                    "You must provide a data source (opendap/netcdf). "
+                    f"Inputs provided: {request.inputs}"
+                )
+        return filepaths
```

Comment:
I believe this method is seen in all of our processes. I think they are all different enough that it may be challenging to factor them all out.  That being said, this one seems to be the most generic of them all, we could probably move it to `utils` and take a look at factoring the others out later.

Reply:
Factoring this one and [the one for `wps_generate_climos.py`](https://github.com/pacificclimate/thunderbird/blob/375fb8aabc01e70409aa3870e5753b4e2b681b10/thunderbird/processes/wps_generate_climos.py#L196) was simple since they're pretty much identical (the other one has a single file rather than multiple files). 

[The one for `wps_generate_prsn.py`](https://github.com/pacificclimate/thunderbird/blob/375fb8aabc01e70409aa3870e5753b4e2b681b10/thunderbird/processes/wps_generate_prsn.py#L131) is more difficult since the input files are in different inputs rather than one. I could change the one for `wps_split_merged_climos.py` to take a list of variables where the inputs are and iterate through each of them, but I'm wondering if that'll get confusing.

[The one for `wps_update_metadata.py`](https://github.com/pacificclimate/thunderbird/blob/375fb8aabc01e70409aa3870e5753b4e2b681b10/thunderbird/processes/wps_update_metadata.py#L69) might also be difficult due to the additional statements in the case where the input is an opendap URL.

Reply:
Ok, I assumed there would be some friction since not all of them were exactly the same.  Factor out the two that you can and leave the others as they are. I don't think it is in the scope of this PR to do too much refactor work anyways.